servers/src/main/java/tachyon/worker/block/TieredBlockStore.java

```diff
@@ -733,16 +735,21 @@ private MoveBlockResult moveBlockInternal(long userId, long blockId,
         srcLocation = srcBlockMeta.getBlockLocation();
         srcFilePath = srcBlockMeta.getPath();
         blockSize = srcBlockMeta.getBlockSize();
+
+        // when the srcLocation equals the dstLocation, simply do nothing and return success
+        if (srcLocation.equals(dstLocation)) {
+          return new MoveBlockResult(true, blockSize, srcLocation, dstLocation);
+        }
       } finally {
         mMetadataReadLock.unlock();
       }
 
-      TempBlockMeta dstTempBlock =
-          createBlockMetaInternal(userId, blockId, newLocation, blockSize, false);
+      TempBlockMeta dstTempBlock = createBlockMetaInternal(userId, blockId, dstLocation,
+          blockSize, false);
       if (dstTempBlock == null) {
         return new MoveBlockResult(false, blockSize, null, null);
       }
-      dstLocation = dstTempBlock.getBlockLocation();
```

Comment:
Actually not entirely true: `newLocation` can have wildcard while `dstLocation` is always a specific tier and dir. I think we were using this because we want to return exact location to caller of this method required by some event listener. I think in the new code, you should change
```java
return new MoveBlockResult(true, blockSize, srcLocation, dstLocation);
``` 
into 
```java
return new MoveBlockResult(true, blockSize, srcLocation, someSpecificBlockLocation);
```